### ISSUE TITLE
HTTP ERROR 404

### ISSUE THREAD

**Issue by ldrrp** [linkedin/cruise-control#556]
HTTP ERROR 404
Problem accessing /kafkacruisecontrol/state. Reason:
    Not Found
Powered by Jetty:// 9.4.6.v20170531

I have followed the setup guide perfectly. But i am getting the following error. My setup is as follows

I have 3 kafka servers running cruise-control
1 server separate running cruise-control-ui as standalone
I left all the ports default
I am using ubuntu 18.04 on all servers.
I setup using git clone method. I am on the latest as of writing this issue.

My cruise-control servers are all reporting 404 not found for any end point in the api.

**Comment by ldrrp** [comment]
I also noticed this setting in cruisecontrol.properties and its not documented in the wiki

webserver.ui.diskpath=./cruise-control-ui/dist/

I dont see that path anywhere on cruise-control

**Comment by nareshv** [comment]
@ldrrp, the config path `webserver.ui.diskpath` is available in the most recent versions of CC. In either case, CC code is aware that when there is a directory like `cruise-control-ui/dist` within CC, then the CC is programmed to serve the frontend from that path when we hit the CC Port.

I have given more details in this ticket https://github.com/linkedin/cruise-control-ui/issues/3. Let us know if you still face any issues.

To summarize

```
git clone https://github.com/linkedin/cruise-control/
cd cruise-control
./gradew jar
..
# build ccfe
git clone https://github.com/linkedin/cruise-control-ui/
cd cruise-control-ui
npm install && npm build
# after this you'll see ./cruise-control-ui/dist folder within CC
# start cc
./kafka-cruise-control-start.sh [params] 8080
# now hitting http://locahost:8080 serves CCFE
```

**Comment by ldrrp** [comment]
@nareshv So each kafka server still has to run cruise control front end? That seems like a waste

**Comment by nareshv** [comment]
@ldrrp. There is no need to run CCFE for every kafka cluster. CCFE can manage all the kafka clusters in your environment. CCFE is designed to work in both standalone (one for all kafka clusters) & embedded mode (one for each kafka cluster)

Here are the brief steps to setup in standalone mode. (Please secure the setup)

```
sudo dnf install httpd mod_ssl
cd $HOME && git clone https://github.com/linkedin/cruise-control-ui/
npm install
# update static/config.csv to have all your CC end point details (see [wiki](https://github.com/linkedin/cruise-control-ui/) for syntax of config.csv)
npm build
rsync -avz dist/* /var/www/html/
```

After this when you hit `http(s)://$host:$port/` you can see CCFE where you can select the kafka cluster that you want to manage.

You need to add few reverse proxy rules to your webserver config so that all requests from the frontend get routed through the webserver.

TODO: i will add a wiki in the next few days to give full examples with few webservers like ATS, nginx, apache.

**Comment by nareshv** [comment]
@ldrrp , i have created wiki on how to deploy CCFE in single or multi kafka cluster environments 
https://github.com/linkedin/cruise-control-ui/wiki/Multiple-Kafka-Clusters
https://github.com/linkedin/cruise-control-ui/wiki/Single-Kafka-Cluster

**Comment by nareshv** [comment]
@ldrrp , let us know if you need any more pointers. Feel free to close the issue in case you were able to get things up and running.
